### PR TITLE
Do not use user defined templateDir and hooksDir

### PR DESCRIPTION
**Pull request by hauleth**
This can cause problems with some user configurations as such flags can
introduce unwanted output/behaviour in some test cases.

### CODE REVIEW

**Comment by josevalim** [comment]
Maybe we should set [`GIT_CONFIG_NOSYSTEM`](https://git-scm.com/docs/git#git-codeGITCONFIGNOSYSTEM) instead?

**Comment by hauleth** [comment]
@josevalim this will only ignore system-wide git configuration, not user-wide one.

File: lib/mix/lib/mix/scm/git.ex [hunk 1]
```diff
@@ -94,8 +94,20 @@ defmodule Mix.SCM.Git do
     File.mkdir_p!(path)
 
     File.cd!(path, fn ->
-      git!(["init", "--quiet"])
-      git!(["--git-dir=.git", "remote", "add", "origin", opts[:git]])
+      git!(~w[-c core.hooksPath='' -c init.templateDir='' init --quiet])
```
Comment (josevalim) [inline lib/mix/lib/mix/scm/git.ex]: I don't think we need to pass `core.hooksPath` here.

**Comment by josevalim** [comment]
> @josevalim this will only ignore system-wide git configuration, not user-wide one.

As the documentation for the variable suggested, we would need to set that and the HOME variable to avoid all configs. But I actually don't think that is a good idea since SSH configuration goes in the config files and we need that. :)

File: lib/mix/lib/mix/scm/git.ex [hunk 2]
```diff
@@ -94,8 +94,20 @@ defmodule Mix.SCM.Git do
...
+
+      git!([
+        "-c",
+        "core.hooksPath=''",
+        "-c",
+        "init.templateDir=''",
```
[the remark this answers is not in the document]
Comment (hauleth) [reply lib/mix/lib/mix/scm/git.ex]: Unfortunately, no. This is also used by `git clone` and I believe that `submodules` use this as well.

File: lib/mix/lib/mix/scm/git.ex (hunk 1 above)
Comment (josevalim) [reply lib/mix/lib/mix/scm/git.ex]: If it isn't needed, then we shouldn't keep it. I don't think consistency here matters. Different commands, different options.

File: lib/mix/lib/mix/scm/git.ex [hunk 3]
```diff
@@ -94,8 +94,20 @@ defmodule Mix.SCM.Git do
     File.mkdir_p!(path)
 
     File.cd!(path, fn ->
-      git!(["init", "--quiet"])
-      git!(["--git-dir=.git", "remote", "add", "origin", opts[:git]])
+      git!(~w[-c core.hooksPath='' -c init.templateDir='' init --quiet])
+
+      git!([
+        "-c",
+        "core.hooksPath=''",
```
Comment (josevalim) [inline lib/mix/lib/mix/scm/git.ex]: Which hooks are invoked on `remote add origin`?

File: lib/mix/lib/mix/scm/git.ex (hunk 2 above)
[the remark this answers is not in the document]
Comment (josevalim) [reply lib/mix/lib/mix/scm/git.ex]: Hrm, I see, thanks. But it certainly isn't used by git-remote, is it?

File: lib/mix/lib/mix/scm/git.ex [hunk 4]
```diff
@@ -94,8 +94,8 @@ defmodule Mix.SCM.Git do
     File.mkdir_p!(path)
 
     File.cd!(path, fn ->
-      git!(["init", "--quiet"])
-      git!(["--git-dir=.git", "remote", "add", "origin", opts[:git]])
+      git!(~w[-c core.hooksPath='' -c init.templateDir='' init --quiet])
+      git!(["remote", "add", "origin", opts[:git]])
```
Comment (josevalim) [inline lib/mix/lib/mix/scm/git.ex]: You removed the --git-dir configuration from this previous command.

File: lib/mix/lib/mix/scm/git.ex (hunk 1 above)
Comment (josevalim) [reply lib/mix/lib/mix/scm/git.ex]: @hauleth and did you look into setting `--templates=""` instead here and below?

**Comment by josevalim** [comment]
Ping!

**Comment by hauleth** [comment]
@josevalim I will finish this later as I had other tasks today (in and out of work). Sorry

**Comment by josevalim** [comment]
Ping :D

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**